ui/src/PrintFulfilment.js

```diff
@@ -0,0 +1,127 @@
+import React, {Component} from 'react';
+import '@fontsource/roboto';
+import {Button, Dialog, DialogContent, FormControl, InputLabel, MenuItem, Select} from '@material-ui/core';
+import {getAllPrintTemplates, getFulfilmentPrintTemplates} from "./Utils";
+
+class PrintFulfilment extends Component {
+  state = {
+    packCode: '',
+    allowableFulfilmentPrintTemplates: [],
+    packCodeValidationError: false,
+    showDialog: false,
+  }
+
+  openDialog = () => {
+    this.refreshDataFromBackend()
+    this.setState({
+      showDialog: true,
+    })
+  }
+
+  closeDialog = () => {
+    this.setState({
+      packCode: '',
+      allowableFulfilmentPrintTemplates: [],
+      packCodeValidationError: false,
+      showDialog: false,
+    })
+  }
+
+  onPrintTemplateChange = (event) => {
+    this.setState({packCode: event.target.value})
+  }
+
+  onCreate = async () => {
+    if (!this.state.packCode) {
+      this.setState({
+        packCodeValidationError: true
+      })
+
+      return
+    }
+
+    const printFulfilment = {
+      "packCode": this.state.packCode
+    }
+
+    const response = await fetch(`/cases/${this.props.caseId}/action/fulfilment`, {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(printFulfilment)
+    })
+
+    if (response.ok) {
+      this.closeDialog()
+    }
+  }
+  
+
+  // TODO: Need to handle errors from Promises
+  refreshDataFromBackend = async () => {
+
+    const allPrintFulfilmentTemplates = await getAllPrintTemplates()
+        .then((templates) => {
+              return templates
+        })
+
+    const fulfilmentPrintTemplates = await getFulfilmentPrintTemplates(this.props.surveyId)
+        .then((fulfilmentTemplates) => {
+          return fulfilmentTemplates
+        })
+
+    let allowableFulfilmentPrintTemplates = []
+    allPrintFulfilmentTemplates
+        .forEach(packCode => {
+            if (fulfilmentPrintTemplates.includes(packCode)) {
+              allowableFulfilmentPrintTemplates.push(packCode)
+            }
+          })
+
+    this.setState({
+      allowableFulfilmentPrintTemplates: allowableFulfilmentPrintTemplates
```

Comment:
why can't you just do: `allowableFulfilmentPrintTemplates: fulfilmentPrintTemplates`? I think this code was originally intended for the "allow" list, to stop people from choosing the same option twice

Reply:
Ah okay that makes sense, thanks